ieeeuoft/newhacks-site-2024
# adding new migrations

## Purpose

- This PR hopefully address the migration issues in the `develop` branch

## Confirmation of changes made 
Note: I ran the command `gh pr diff 12` to confirm what changes are made in this PR 

1. Deleted the old 0008 migration file. Exact file name deleted: `0008_auto_20240827_1738.py`

2. Manually made a new 0008 migration file (name is called: 0008_revert_0007_changes.py). Purpose: Reverting the changes made by migration file: `0007_auto_20240826_2200.py`AND reapplies the intended changes in `0008_auto_20240827_1738.py`  and `0007_auto_20240826_2200.py` , but consolidated into 1 file.

Note: In an earlier commit, these changes were made in 2 separate files. However, applying these changes separately would mean there is a risk of 1 migration file being run but not the other (causing the site to go down).

Since migration files are atomic, combining all intended changes into 1 file would mitigate this risk. If this new migration file fails to be applied, then it would not be applied, and the site would retain its current state. 

## Steps to QA

Stage 1: Pull this PR, and see whether the local site (http://127.0.0.1:8000/) as well as the admin site (http://127.0.0.1:8000/admin) compiles without error 

Stage 2: Try recreating the application of this migration manually. What you can do is to checkout into a new branch (from `develop`). If you try running `python manage.py migrate` as it currently stands, you should get the `ALTER TABLE` error seen in the develop branch. 

Now, you can try manually adding back this new migration file in the directory. Do `python manage.py showmigrations` to show the list of migrations (and make sure the new migration file is listed). Then, try doing `python manage.py migrate`. The website should work with no issues (no data loss either) AND Django should not prompt you to make additional migrations.
